### PR TITLE
Split up the minify preset

### LINKED ISSUE #807: The library preset enables all types of minification by default
The library preset `.use()`s the minify preset without passing any options, which activates all of {babel-minify, style-minify, image-minify}:
https://github.com/mozilla-neutrino/neutrino-dev/blob/ce7139784507f788ad8509ef5cbc0ad30da34f36/packages/library/index.js#L150

Compare this to the web preset:
https://github.com/mozilla-neutrino/neutrino-dev/blob/ce7139784507f788ad8509ef5cbc0ad30da34f36/packages/web/index.js#L213

Issues:
1) I'm not sure all types of minification be enabled by default. Thoughts?
2) We should allow people to adjust the options, like we do for the web preset.

Filing this since it affects the current implementation of the webpack-4 branch, since that uses (at least for now, to let us A/B test) a toggle to switch between uglify-es and babel-minify.

### PR DESCRIPTION
**Pull request by edmorley**
* Removes the web preset's dependency on `@neutrinojs/image-minify` since it has many heavyweight native code dependencies, and isn't enabled by default anyway (see #713).
* Removes the library preset's dependency on `@neutrinojs/style-minify` and `@neutrinojs/image-minify` (fixes #807).
* Removes the `@neutrinojs/minify` preset in favour of directly depending on the individual `@neutrinojs/*-minify` presets.
* Adds an `enabled` option to `@neutrinojs/image-minify` that defaults to production only, to allow for string form usage of the preset in `.neutrinorc.js`, without the need to specify a `.when()`.

### CODE REVIEW

File: packages/image-minify/test/middleware_test.js [hunk 1]
```diff
@@ -2,32 +2,42 @@ import test from 'ava';
 import { Neutrino } from 'neutrino';
 
 const mw = () => require('..');
+const prodEnv = { 'env': { NODE_ENV: 'production' } };
```
Comment (eliperelman) [inline packages/image-minify/test/middleware_test.js]: We can remove the string from the `env` key.

File: packages/image-minify/test/middleware_test.js [hunk 2]
```diff
@@ -2,32 +2,42 @@ import test from 'ava';
...
+const devEnv = { 'env': { NODE_ENV: 'development' } };
```
Comment (eliperelman) [inline packages/image-minify/test/middleware_test.js]: We can remove the string from the `env` key.

**Comment by eliperelman** [comment]
I think this also fixes #670 in master.

**Comment by edmorley** [comment]
Though presumably anyone who explicitly adds the `@neutrinojs/image-minify` preset will still be affected?

**Comment by eliperelman** [comment]
Correct.

**Pull request merged**